src/components/app.tsx

```diff
@@ -230,8 +256,13 @@ export class App extends React.PureComponent<IProps, IState> {
   }
 
   private handleChangePage = (page: number) => {
-    this.setState({currentPage: page});
-    setDocumentTitle(this.state.activity, page);
+    if (page > this.state.currentPage && this.state.incompleteQuestions.length > 0) {
+      const label = this.state.incompleteQuestions[0].navOptions?.message || kDefaultIncompleteMessage;
+      this.setShowModal(true, label);
+    } else if (page >= 0 && (this.state.activity && page <= this.state.activity.pages.length)) {
+      this.setState({currentPage: page, incompleteQuestions: []});
+      setDocumentTitle(this.state.activity, page);
+    }
```

Comment:
You could get rid of a bunch of `this.state`s above with:
```typescript
  const { currentPage, incompleteQuestions, activity } = this.state;
```

Reply:
Fixed.